authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java

```diff
@@ -232,74 +148,172 @@ public SecurityFilterChain loginAuthSecurityFilterChain(
+          .exceptionHandling(
+              // this prevents the usage of the default BasicAuthenticationEntryPoint returning
+              // a WWW-Authenticate header that causes browsers to prompt for basic login
+              exceptionHandling ->
+                  exceptionHandling
+                      .authenticationEntryPoint(authFailureHandler)
+                      .accessDeniedHandler(authFailureHandler))
+          // do not create a session on api authentication, that's to be done on webapp login only
+          .sessionManagement(
+              (sessionManagement) ->
+                  sessionManagement.sessionCreationPolicy(SessionCreationPolicy.NEVER))
+          .build();
+    }
 
-  private static HttpSecurity withSecurityMatcher(
-      final HttpSecurity httpSecurity, final RequestMatcher requestMatcher) {
-    return httpSecurity.securityMatchers(matchers -> matchers.requestMatchers(requestMatcher));
+    @Bean
+    @Order(ORDER_WEBAPP_API)
+    public SecurityFilterChain httpBasicWebappAuthSecurityFilterChain(
+        final HttpSecurity httpSecurity,
+        final AuthFailureHandler authFailureHandler,
+        final WebApplicationAuthorizationCheckFilter webApplicationAuthorizationCheckFilter)
+        throws Exception {
+      LOG.info("Web Applications Login/Logout is setup.");
+      return httpSecurity
+          .securityMatcher(WEBAPP_PATHS.toArray(String[]::new))
+          // webapps are accessible without any authentication required
+          .authorizeHttpRequests(
+              (authorizeHttpRequests) -> authorizeHttpRequests.anyRequest().permitAll())
+          .csrf(AbstractHttpConfigurer::disable)
+          .cors(AbstractHttpConfigurer::disable)
+          .anonymous(AbstractHttpConfigurer::disable)
+          // http basic auth is possible to obtain a session
+          .httpBasic(Customizer.withDefaults())
+          // login/logout is still possible to obtain a session
+          // the session grants access to the API as well, via #httpBasicApiAuthSecurityFilterChain
+          .formLogin(
+              formLogin ->
+                  formLogin
+                      .loginPage(LOGIN_URL)
+                      .loginProcessingUrl(LOGIN_URL)
+                      .failureHandler(authFailureHandler)
+                      .successHandler(WebSecurityConfig::noContentSuccessHandler))
+          .logout(
+              (logout) ->
+                  logout
+                      .logoutUrl(LOGOUT_URL)
+                      .logoutSuccessHandler(WebSecurityConfig::noContentSuccessHandler)
+                      .deleteCookies(SESSION_COOKIE))
+          .exceptionHandling(
+              exceptionHandling ->
+                  exceptionHandling
+                      .authenticationEntryPoint(authFailureHandler)
+                      .accessDeniedHandler(authFailureHandler))
+          .addFilterAfter(webApplicationAuthorizationCheckFilter, AuthorizationFilter.class)
```

Comment:
@maryarm with this dedicated chain setup I could indeed just add the authorization filter to the webapp security chain